src/Workspaces/Core/Portable/CodeFixes/FixAllOccurrences/DefaultFixAllProviderHelpers.cs

```diff
@@ -74,5 +75,28 @@ internal static class DefaultFixAllProviderHelpers
                 fixAllContext,
                 sortedProjects.SelectAsArray(p => fixAllContext.WithScope(FixAllScope.Project).WithDocumentAndProject(document: null, p)));
         }
+
+        private static async Task<Solution?> GetContainingMemberOrTypeFixesAsync(FixAllContext fixAllContext, FixAllContexts fixAllContextsAsync)
+        {
+            Contract.ThrowIfNull(fixAllContext.Document);
+
+            if (!fixAllContext.State.TriggerSpan.HasValue)
+                return null;
+
+            var spanMappingService = fixAllContext.Document.GetLanguageService<IFixAllSpanMappingService>();
+            if (spanMappingService == null)
+                return null;
+
+            var documentsAndSpansToFix = await spanMappingService.GetFixAllSpansAsync(
+                fixAllContext.Document, fixAllContext.State.TriggerSpan.Value,
+                fixAllContext.Scope, fixAllContext.CancellationToken).ConfigureAwait(false);
+            if (documentsAndSpansToFix.IsEmpty)
+                return null;
+
+            var fixAllContexts = documentsAndSpansToFix.SelectAsArray(
+                (documentAndSpans) => fixAllContext.WithDocumentAndProject(documentAndSpans.Key, documentAndSpans.Key.Project)
```

Comment:
asides: WithDoucmentAndProject feels somewhat weird given that a project is inferrable from a DOcument.  
also, i'm nto a fan of all the separate With'ers.  Could just have one and do `With(document: docAndSpan.Key, fixAllSpans: docAndSpan.Value)`

Reply:
There are existing callsites for this API that pass a null document and non-null project: https://github.com/dotnet/roslyn/blob/7e835c8c324e294b467ba69648a2049e3acbcc6f/src/Workspaces/Core/Portable/CodeFixes/FixAllOccurrences/DefaultFixAllProviderHelpers.cs#L51 and https://github.com/dotnet/roslyn/blob/7e835c8c324e294b467ba69648a2049e3acbcc6f/src/Workspaces/Core/Portable/CodeFixes/FixAllOccurrences/DefaultFixAllProviderHelpers.cs#L75

> i'm nto a fan of all the separate With'ers

I agree, but going to leave this as is to avoid unrelated stylistic API changes in this PR.